### DISCUSSION TITLE
Entities do not exists yet in PreStartup/Startup/PostStartup stage

### DISCUSSION THREAD

**Discussion by Twyker** [Trouv/bevy_ecs_ldtk#260] [Q&A] [answered]
Hi! 
It seems entities imported from LDtk are being spawned after being registered, but sadly only exists first at the Update stage, not in any of the Startup stages.
This is really hindering normal development as we can't use the data from LDtk for any kind of one-off/init logic.
**Is there any way to access the entities before Update? When are they being spawned/created? What's the best way to go about this?** 🙏 😅 

Thanks for any help/assistance 🥺

**Comment by Trouv** [comment, marked as answer]
This is true, you won't have access to these things during the Startup schedule This is actually just the nature of bevy's asset system being asynchronous, not really the design of this plugin. When loading an asset, you'll only get a handle to that asset which will eventually point to the asset's data in its asset store, but it will point to nothing for some time. So, if you're loading assets on startup, that data won't be available for at least one update after startup.

If you need to do some processing on asset data once after it is loaded, listening for that asset's `AssetEvent` in a system is a pretty common strategy. Since `bevy_ecs_ldtk` does its own processing on `.ldtk` assets, it also provides other mechanisms like [`LevelEvent`](https://docs.rs/bevy_ecs_ldtk/0.8.0/bevy_ecs_ldtk/enum.LevelEvent.html). For entities, you can have a system that queries for `Added<EntityInstance>`, or preferably (assuming you've inserted some custom components on it via `LdtkEntity` registration), `Added<MyComponent>`. These queries will only contain entities during the update that `bevy_ecs_ldtk` spawned them, and systems running on empty queries don't have much overhead. If you really want to limit this overhead, you can try using bevy run conditions to have them only run if such entities exist, or only run on `LevelEvent`s, or only run in a custom game state responsible for level transitions/spawning.

I also highly recommend using [`bevy_asset_loader`](https://crates.io/crates/bevy_asset_loader) for managing state transitions related to asset loading in general. Not necessarily applicable for running systems once for LDtk entity spawns since the plugin does that on its own time, but it is a generally useful crate for handling the "I want to do something once when an asset is loaded" problem.

**Reply by Twyker** [reply]
Hm ok, might try it with Added<X> first, although it still doesn't feel great to to initialization logic at Update. ^^

Do you think the new one shot systems might help with this? :)
